demos/python_demos/object_detection_demo_centernet/detector.py

```diff
@@ -0,0 +1,217 @@
+            new_pt = np.dot(t, new_pt)
+            return new_pt[:2]
+
+        target_coords = np.zeros(coords.shape)
+        trans = Detector.get_affine_transform(center, scale, 0, output_size, inv=1)
+        for p in range(coords.shape[0]):
+            target_coords[p, 0:2] = affine_transform(coords[p, 0:2], trans)
+        return target_coords
+
+    @staticmethod
+    def _transform(dets, center, scale, height, width):
+        dets[:, :2] = Detector._transform_preds(
+            dets[:, 0:2], center, scale, (width, height))
+        dets[:, 2:4] = Detector._transform_preds(
+            dets[:, 2:4], center, scale, (width, height))
+        return dets
+
+    def preprocess(self, image):
+        height, width = image.shape[0:2]
+        center = np.array([width / 2., height / 2.], dtype=np.float32)
+        scale = max(height, width) * 1.0
+
+        trans_input = self.get_affine_transform(center, scale, 0, [self.input_width, self.input_height])
+        resized_image = cv2.resize(image, (width, height))
+        inp_image = cv2.warpAffine(
+            resized_image, trans_input, (self.input_width, self.input_height),
+            flags=cv2.INTER_LINEAR)
+
+        image = inp_image
+        return image
+
+    def postprocess(self, raw_output, image_sizes):
+        heat, reg, wh = raw_output
+        heat = 1/(1 + np.exp(-heat))
+        height, width = heat.shape[1:3]
+
+        heat = self._nms(heat)
+        scores, inds, clses, ys, xs = self._topk(heat, K=100)
+        reg = self._tranpose_and_gather_feat(reg, inds)
+        num_predictions = 100
+
+        reg = reg.reshape((num_predictions, 2))
+        xs = xs.reshape((num_predictions, 1)) + reg[:, 0:1]
+        ys = ys.reshape((num_predictions, 1)) + reg[:, 1:2]
+
+        wh = self._tranpose_and_gather_feat(wh, inds)
+        wh = wh.reshape((num_predictions, 2))
+        clses = clses.reshape((num_predictions, 1)).astype(np.float)
+        scores = scores.reshape((num_predictions, 1))
+        bboxes = np.concatenate((xs - wh[..., 0:1] / 2,
+                                 ys - wh[..., 1:2] / 2,
+                                 xs + wh[..., 0:1] / 2,
+                                 ys + wh[..., 1:2] / 2), axis=1)
+        detections = np.concatenate((bboxes, scores, clses), axis=1)
+        mask = detections[..., 4] >= self._threshold
+        filtered_detections = detections[mask]
+        scale = max(image_sizes)
+        center = np.array(image_sizes[:2])/2.0
+        dets = self._transform(filtered_detections, np.flip(center), scale, height, width)
+        return dets
+
+    def infer(self, image):
+        t0 = cv2.getTickCount()
+        output = self._exec_model.infer(inputs={self._input_layer_name: image})
+        self.infer_time = (cv2.getTickCount() - t0) / cv2.getTickFrequency()
+        return output
+
+    def detect(self, image):
+        image_sizes = image.shape[:2]
+        image = self.preprocess(image)
+        if image.shape[2] == 3:
```

Comment:
If you expect that that image dimensions can come in a different order, you should conduct this check earlier, because `image_sizes = image.shape[:2]` and `image = self.preprocess(image)` depend on it. But I think you just shouldn't expect, that the order can be different and remove the check.